agent/exploits/cve_2023_43770.py

```diff
@@ -0,0 +1,137 @@
+"""Agent Asteroid implementation for CVE-2023-43770"""
+import re
+from urllib import parse as urlparse
+
+import requests
+from ostorlab.agent.kb import kb
+from ostorlab.agent.mixins import agent_report_vulnerability_mixin
+from packaging import version
+from requests import exceptions as requests_exceptions
+
+from agent import definitions
+from agent import exploits_registry
+
+VULNERABILITY_TITLE = (
+    "Roundcube Webmail Persistent Cross-Site Scripting (XSS) Vulnerability"
+)
+VULNERABILITY_REFERENCE = "CVE-2023-43770"
+VULNERABILITY_DESCRIPTION = (
+    "Roundcube Webmail contains a persistent cross-site scripting (XSS) vulnerability that can "
+    "lead to information disclosure via malicious link references in plain/text messages."
+)
+
+DEFAULT_TIMEOUT = 30
+MAIL_ENDPOINT_PATTERN = re.compile(r'content="\d; url=(.*)"')
+DEFAULT_MAIL_ENDPOINT = "/mail/"
+VERSION_PATTERN = re.compile(r'"rcversion":(\d+)')
+VULNERABLE_RANGES = [
+    (version.parse("1.0.0"), version.parse("1.4.13")),
+    (version.parse("1.5.0"), version.parse("1.5.3")),
+    (version.parse("1.6.0"), version.parse("1.6.2")),
+]
+MAX_REDIRECTS = 2
+ROUNDCUBE_KEYWORD = "rcmail"
+
+
+@exploits_registry.register
+class CVE202343770Exploit(definitions.Exploit):
+    """
+    CVE-2023-43770: Roundcube Webmail Persistent Cross-Site Scripting (XSS) Vulnerability
+    """
+
+    def __init__(self) -> None:
+        session = requests.Session()
+        session.max_redirects = MAX_REDIRECTS
+        session.verify = False
+        self.session = session
```

Comment:
We keep writing the same code. Please write a WebExploit that have these and avoid repeating the same code over and over.